Makefile

```diff
@@ -209,6 +209,38 @@ check-kuttl:
 	${PGO_KUBE_CLIENT} ${KUTTL_TEST} \
 		--config testing/kuttl/kuttl-test.yaml
 
+# KUTTL_ params -- could be left out entirely and just set as env vars
+KUTTL_PSQL_VERSION ?= 14
+# TODO: should the test image here be set according to the KUTTL_PSQL_VERSION
+KUTTL_PSQL_IMAGE ?= registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-0
+
+# TODO: When I attempted to loop through the directories and find the yaml files,
+# due to how makefile expands variables, each file would end up in each directory.
+# So this implementation loops through to create the folders under generated
+# and then loops through all the yaml files, using the dirname/basename to place
+# the generated file in the correct folder
+.PHONY: generate-kuttl-general
+generate-kuttl-general: 
+	for SRCDIR in $(shell ls testing/kuttl/source); do \
+		KUTTL_PSQL_VERSION=$(KUTTL_PSQL_VERSION) \
+			mkdir -p testing/kuttl/generated/$(KUTTL_PSQL_VERSION)-$${SRCDIR}; \
+	done;
+	for FILENAME in $(shell find testing/kuttl/source -regex ".*\.yaml" -print); do \
+		KUTTL_PSQL_VERSION=$(KUTTL_PSQL_VERSION) \
+			KUTTL_PSQL_IMAGE=$(KUTTL_PSQL_IMAGE) \
+			envsubst < $${FILENAME} \
+				> testing/kuttl/generated/$(KUTTL_PSQL_VERSION)-$$(basename $$(dirname $${FILENAME}))/$$(basename $${FILENAME}); \
+	done
+
+# TODO: Generate upgrade KUTTL tests might require different handling
```

Comment:
🤔 It seems like upgrade tests need two versions. Can we have two variables? `_VERSION` and `_UPGRADE_FROM_VERSION` ?

Reply:
tested and it looks fine to me.